contrib/filter-repo-demos/timeshift.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import subprocess
+import datetime
+import sys
+try:
+  import git_filter_repo as fr
+except ImportError:
+  raise SystemExit("Error: Couldn't find git_filter_repo.py.  Did you forget to make a symlink to git-filter-repo named git_filter_repo.py or did you forget to put the latter in your PYTHONPATH?")
+
+parser = argparse.ArgumentParser(
+          description='Offset a set of or all commits by n seconds, can be used to timeshift a repository')
+parser.add_argument('--offset', type=int,
+        help=("Offset in seconds"))
+parser.add_argument('--from-commit', type=str,
+        help=("Commit to start from"))
+parser.add_argument('--to-commit', type=str,
+        help=("Commit to finish at"))
+args = parser.parse_args()
+
+print(args)
+
+shouldOffset = args.from_commit is None
+
+def offsetDate(timestampAndUtcOffset):
+  timestampAndUtcOffset = timestampAndUtcOffset.decode('utf-8')
+  split = timestampAndUtcOffset.split(' ')
+  timestamp = int(split[0])
+  timestamp = timestamp + args.offset
+  result = str(timestamp) + ' ' + split[1];
+  result = result.encode()
+  return result
+
+def fixup_commits(commit, metadata):
+  global shouldOffset
+  commitId = commit.original_id.decode('utf-8')
```

Comment:
decoding to utf-8 is kinda slow, even when it's already known to be ascii.  It'd be easier/faster to encode args.from_commit and args.to_commit -- especially since that only has to be done once instead of one decoding per commit in the repository.  In fact, you just need to add type=os.fsencode to the parser.add_argument() call and then it'll show up as bytes that you can compare directly.

Reply:
Noted, wasn't aware of the os.fsencode type